util/files/util.go

```diff
@@ -0,0 +1,64 @@
+package files
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+var RelativeOutOfBoundErr = errors.New("full path does not contain base path")
+
+// RelativePath will remove the basePath string from the fullPath
+// including the path separator. Differently from filepath.Rel, this
+// function will return error (RelativeOutOfBoundErr) if basePath
+// does not match (example 2).
+//
+// Example 1:
+//   fullPath: /home/test/app/readme.md
+//   basePath: /home/test
+//   return:   app/readme.md
+//
+// Example 2:
+//   fullPath: /home/test/app/readme.md
+//   basePath: /somewhere/else
+//   return:   "", RelativeOutOfBoundErr
+//
+// Example 3:
+//   fullPath: /home/test/app/readme.md
+//   basePath: /home/test/app/readme.md
+//   return:   .
+func RelativePath(fullPath, basePath string) (string, error) {
+	fp := filepath.Clean(fullPath)
+	if !strings.HasPrefix(fp, basePath) {
```

Comment:
```suggestion
	if !strings.HasPrefix(fp, filepath.Clean(basePath)) {
```

This will handle the unlikely scenario of basePath containing relative components. For example, we'd want `fp = /b` to match `basePath = /a/../b`.

The same logic is in util/app/path/path.go, in case you care to abstract that out.

Reply:
done